lib/posix/options/fs.c

```diff
@@ -157,6 +166,15 @@ static int fs_ioctl_vmeth(void *obj, unsigned int request, va_list args)
 		}
 		break;
 	}
+#if defined(CONFIG_POSIX_FILE_SYSTEM_FSTAT)
+	case ZFD_IOCTL_STAT: {
+		struct stat *buf = NULL;
+
+		buf = va_arg(args, struct stat *);
+		rc = stat(ptr->path, buf);
```

Comment:
Based on my comments above, I think that the algorithm here is possibly backwards.

for `stat()`, given the absolute path of a file, we should

1. convert that to a tuple (mount point, node), where node is a pointer to an fs-specific handle, even just an offset into storage.
2. Use that tuple to call a file-system-specific method that returns data about the node.

For `fstat()`, we should
1. determine the mount point to generate the (mount point, node) tuple
2. Use that tuple to call the same file-system-specific method that returns data about the node.